bson-kotlinx/src/test/kotlin/org/bson/codecs/kotlinx/KotlinSerializerCodecProviderTest.kt

```diff
@@ -60,4 +72,34 @@ class KotlinSerializerCodecProviderTest {
         assertTrue { codec is KotlinSerializerCodec }
         assertEquals(DataClassWithSimpleValues::class.java, codec.encoderClass)
     }
+
+    @OptIn(ExperimentalSerializationApi::class)
+    @Test
+    fun shouldAllowOverridingOfSerializersModuleAndBsonConfigurationInConstructor() {
+
+        val serializersModule =
+            SerializersModule {
+                this.polymorphic(DataClassOpen::class) {
+                    this.subclass(DataClassOpenA::class)
+                    this.subclass(DataClassOpenB::class)
+                }
+            } + defaultSerializersModule
+
+        val bsonConfiguration = BsonConfiguration(classDiscriminator = "__type")
+        val dataClassContainsOpenB = DataClassContainsOpen(DataClassOpenB(1))
+
+        val codec =
+            KotlinSerializerCodecProvider(serializersModule, bsonConfiguration)
+                .get(DataClassContainsOpen::class.java, Bson.DEFAULT_CODEC_REGISTRY)!!
+
+        assertTrue { codec is KotlinSerializerCodec }
+        val encodedDocument = BsonDocument()
+        val writer = BsonDocumentWriter(encodedDocument)
+        codec.encode(writer, dataClassContainsOpenB, EncoderContext.builder().build())
+        writer.flush()
+
+        assertEquals(
+            BsonDocument.parse("""{"open": {"__type": "org.bson.codecs.kotlinx.samples.DataClassOpenB", "b": 1}}"""),
```

Comment:
[optional]
We may also test that decoding this BSON succeeds and produces an object equal to `dataClassContainsOpenB`. This way we'll know that the custom `classDiscriminator` is used not only by the encoder, but also by the decoder.

Reply:
Done